.devcontainer/devcontainer.json

```diff
@@ -1,14 +1,26 @@
 {
   "name": "AWS CDK & Python Development Environment",
   "image": "mcr.microsoft.com/devcontainers/base:ubuntu-22.04",
+  "containerEnv": {
```

Comment:
without documentation i don't know how this works relative to the env vars from the CI/CD workflow

Reply:
This is something that should be transparent to the user and so I don't see the need to document that change. This change is only relevant in the context of running the dev container inside a GitHub workflow.

Reply:
When starting a dev container with that definition, if a given env var exist on the host, this env var (KV) will be created in the dev container.

Reply:
> This change is only relevant in the context of running the dev container inside a GitHub workflow.

hmm maybe i don't understand the use case of running the dev container inside of a Github workflow.  What is your intention for this?

Reply:
This PR is adding Docker-in-Docker to the dev container so that we can still run the command `cdk deploy --all --concurrency 5 --require-approval never` inside the dev container. We also need to run that command in the GH workflow, so the workflow environment needs to provide the `docker` command. I now realize that the GH hosted runner should have Docker available. The alternative was to run the `cdk ...` command inside the dev container in the workflow. The additional benefit is that the dev container comes will all the development tools so that the following steps could be removed from the workflow:

```yaml
      - name: Install AWS CLI
        run: sudo snap install aws-cli --classic
      - name: Install AWS CDK CLI
        run: npm install -g aws-cdk
```

If we want to run the command `cdk ...` inside the dev container inside the workflow, then the dev container needs to have access to the "output" of the action `aws-actions/configure-aws-credentials@v2`. My understanding is that this output are the `AWS_*` environment variables. Here is summary of what happens in the workflow:

1. The action `aws-actions/configure-aws-credentials@v2` set the `AWS_*` env vars on the host (GitHub runner).
2. We start the dev container. Because the `AWS_*` env vars are defined in the `devcontainer.json`, these env vars that exist now on the host/runner will be available inside the dev container.
3. We run `cdk ...` inside the dev container. This command will be authorized to deploy the CDK app because the `AWS_*` are set.
1. The work